### PR TITLE
Add more startup logging

### PR DESCRIPTION
**Pull request by gabrieljackson**
This change increases server startup logging by reporting a few config values that may be valuable for troubleshooting.

Fixes https://mattermost.atlassian.net/browse/MM-49620

### CODE REVIEW

File: internal/server/server.go [hunk 1]
```diff
@@ -70,7 +70,8 @@ func New(cfg config.Config) (*Server, error) {
 		keyDataMap: make(map[pgproto3.BackendKeyData]*ClientConn),
 	}
 
-	s.logger.Info("Initializing server...")
+	s.logger.Info("Initializing server...", logr.String("log-level", cfg.LogSettings.Level), logr.Bool("json-logging", cfg.LogSettings.Json))
```
Comment (agnivade) [inline internal/server/server.go:73]: The logsettings part was resolved right? Is there still any benefit of adding these? I am thinking later we might add an endpoint to return the config itself. That way the user can confirm what config values are in effect.

Comment (gabrieljackson) [reply internal/server/server.go:73]: I do believe there is value in these logs still. Primarily, these sorts of log lines are common in our cloud services as a way to verify that configuration has been properly updated across all replicas in all accounts when we roll pods. Having a way to query the config in a more robust way is nice, but logs help us verify that the fleet is updating and working as expected historically. Only 1/5 on getting this in though if you want to keep log noise down to an absolute minimum.

Comment (agnivade) [reply internal/server/server.go:73]: I am okay with the ListenAddress and MetricAddress, but the logSettings feel a bit arbitrary to me. Why not poolSettings as well?

Comment (gabrieljackson) [reply internal/server/server.go:73]: Yeah, valid question. Log settings was the most useful at this moment. I will update this to just include the addresses for now and we can revisit later.

**Review by agnivade — APPROVED** [review]

File: internal/server/server.go [hunk 2]
```diff
@@ -71,6 +71,7 @@ func New(cfg config.Config) (*Server, error) {
 	}
 
 	s.logger.Info("Initializing server...")
+	s.logger.Info("Initializing database proxy listener...", logr.String("listen-address", cfg.ListenAddress))
```
Comment (agnivade) [inline internal/server/server.go:74]: Nit: Maybe move `logr.String("listen-address", cfg.ListenAddress)` as part of the first log line itself? "server" and "database proxy listener" kinda mean the same thing.

**Comment by agnivade** [comment]
Good to go, thanks!

**Pull request merged**